tests/components/openweathermap/test_sensor.py

```diff
@@ -0,0 +1,71 @@
+"""Tests for OpenWeatherMap sensors."""
+
+import pytest
+from syrupy import SnapshotAssertion
+
+from homeassistant.components.openweathermap.const import (
+    OWM_MODE_FREE_CURRENT,
+    OWM_MODE_FREE_FORECAST,
+    OWM_MODE_V30,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import mock_config_entry
+from .test_config_flow import _create_static_weather_report
+
+from tests.common import AsyncMock, patch
+
+# Define test data for mocked weather report
+static_weather_report = _create_static_weather_report()
+
+# Define test parameters with different modes and corresponding mock return values
+TEST_CASES = [
+    (
+        OWM_MODE_FREE_CURRENT,
+        "pyopenweathermap.client.free_client.OWMFreeClient.get_weather",
+        static_weather_report,
+    ),
+    (
+        OWM_MODE_FREE_FORECAST,
+        "pyopenweathermap.client.free_client.OWMFreeClient.get_weather",
+        static_weather_report,
+    ),
+    (
+        OWM_MODE_V30,
+        "pyopenweathermap.client.onecall_client.OWMOneCallClient.get_weather",
+        static_weather_report,
+    ),
+]
```

Comment:
This can be directly added to the parameterized decorator. No need to define a constant for this.